rdf-delta-server-local/src/main/java/org/seaborne/delta/server/local/patchstores/zk/PatchLogIndexZk.java

```diff
@@ -92,127 +69,40 @@
      * <li>{@code versionsPath} - directory where the meta data JSON object is stored as name "%08d"
      * </ul>
      */
-
-    private static AtomicInteger counter = new AtomicInteger(0);
-
-    public PatchLogIndexZk(CuratorFramework client, String instance, DataSourceDescription dsd, String logPath) {
+    public PatchLogIndexZk(UncheckedZkConnection client, String instance, DataSourceDescription dsd, String logPath) {
         // THis gets called twice in the creator - sees its own create via ZK watcher.
         this.client = client ;
-        this.instance = instance;
-        this.dsd = dsd;
         this.logName = dsd.getName();
-        this.statePath      = zkPath(logPath, ZkConst.nState);
-        this.lockPath       = zkPath(logPath, ZkConst.nLock);
-        this.lockStatePath  = zkPath(logPath, ZkConst.nLockState);
-        this.versionsPath   = zkPath(logPath, ZkConst.nVersions);
-        this.headersPath    = zkPath(logPath, ZkConst.nHeaders);
+        this.statePath      = ZKPaths.makePath(logPath, ZkConst.nState, new String[]{});
+        this.lockPath       = ZKPaths.makePath(logPath, ZkConst.nLock, new String[]{});
+        this.lockStatePath  = ZKPaths.makePath(logPath, ZkConst.nLockState, new String[]{});
+        this.versionsPath   = ZKPaths.makePath(logPath, ZkConst.nVersions, new String[]{});
+        this.headersPath    = ZKPaths.makePath(logPath, ZkConst.nHeaders, new String[]{});
         this.logStateWatcher = (event)->{
           synchronized(lock) {
               FmtLog.debug(LOG, "++ [%s:%s] Log watcher", instance, logName);
               syncState();
           }
         };
-        this.zkLock = Zk.zkCreateLock(client, lockPath);
 
         // Find earliest.
-        List<String> x = Zk.zkSubNodes(client, versionsPath);
+        List<String> x = client.fetchChildren(versionsPath);
         //Guess: 1
         if ( x.isEmpty() )
             earliestVersion = Version.INIT;
         else if ( x.contains("00000001") )
             // Fast-track the "obvious" answer
             earliestVersion = Version.create(1);
         else {
-            try {
-                long ver = x.stream().map(this::versionFromName).filter(v->(v>0)).min(Long::compare).get();
-                earliestVersion = Version.create(ver);
-            } catch (NoSuchElementException ex) {  }
+            x.stream()
+                .map(this::versionFromName)
+                .filter(v->(v>0))
+                .min(Long::compare)
+                .ifPresent(version -> earliestVersion = Version.create(version));
         }
```

Comment:
It's [functionally equivalent](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#ifPresent-java.util.function.Consumer-) because you're swallowing the exception without doing anything at all with it. Moreover, it now fits the recommendations of the team that created the Streams API.

Reply:
That being said, this is unrelated to the problem I'm trying to solve with this PR, so I'll back this out.